File: Patterns/half-pyramid.cpp
#include<iostream>
using namespace std;

int main(){
  int number;
  cout << "Enter the number: ";
  cin >> number;

  for(int row = 0; row < number; row++){
    for(int col = 0; col < row + 1; col++){
      cout << "* ";
    }
    cout << endl;
  }
}

File: Patterns/half-pyramid.py

""" 
Half Pyramid
*
* *
* * *
* * * *
* * * * *
"""


# Approach 1
for row in range(9):
  for col in range(row+1):
    print("*", end=" ")
  print()  
  
  
# Approach 2
for row in range(7):
  print("* " * (row + 1))


# Approach 3
for row in range(1, 8):
  print(" ".join(["*"] * row))